storage/operation/reads.go

```diff
@@ -206,24 +206,14 @@ func FindHighestAtOrBelowByPrefix(r storage.Reader, prefix []byte, height uint64
 	}
 
 	key := append(prefix, EncodeKeyPart(height)...)
-	it, err := r.NewIter(prefix, key, storage.DefaultIteratorOptions())
-	if err != nil {
-		return fmt.Errorf("can not create iterator: %w", err)
-	}
-	defer func() {
-		errToReturn = merr.CloseAndMergeError(it, errToReturn)
-	}()
 
-	var highestKey []byte
+	seeker := r.NewSeeker()
 
-	// find highest value below the given height
-	for it.First(); it.Valid(); it.Next() {
-		// copy the key to avoid the underlying slices of the key
-		// being modified by the Next() call
-		highestKey = it.IterItem().KeyCopy(highestKey)
+	highestKey, found, err := seeker.SeekLE(prefix, key, key)
```

Comment:
I think it's better to use `prefix` as endPrefix, instead of using `key` as prefix, which would be 
a bit confusing.

```suggestion
	// seeking the highest key equal or below to the given key within the [prefix, prefix] prefix range
	highestKey, found, err := seeker.SeekLE(prefix, prefix, key)
```

Reply:
If we do that, then `key` would be outside range of `[prefix, prefix]`.  `SeekLE` expects `key` to be in range of `startPrefix` and `endPrefix`.

I updated comment here to mention that `endPrefix` is user-specified height (key) because we are interested in any key equal to or less than user-specified height here.

```Go
// Seek the highest key equal or below to the given key within the [prefix, key] prefix range
// Note: We are only interested in key at or less than given height.
endPrefix := key // endPrefix is the caller-specified height (key).
highestKey, found, err := seeker.SeekLE(prefix, endPrefix, key)
```